### PR TITLE
Expose fields in AggregateElements and add integration tests

### PR DESCRIPTION
**Pull request by zyiou**
Users cannot create AggregateElements in AggregationInput because elements are not exposed as public. 
Added integration tests to ensure everything work well.
Also refactored test cases in exporter-collector tests.

### CODE REVIEW

**Review by srikartati — DISMISSED** [review]
Thanks for extending the test to integrate tests too. Yes it's essential to make those fields public. Missed it.

File: pkg/test/exporter_collector_test.go [hunk 1]
```diff
@@ -260,172 +283,246 @@ func testExporterToCollector(address net.Addr, isMultipleRecord bool, isEncrypte
...
 		templateID := export.NewTemplateID()
-		templateSet := entities.NewSet(entities.Template, templateID, false)
-		elements := make([]*entities.InfoElementWithValue, 0)
-		element, err := registry.GetInfoElement("sourceIPv4Address", registry.IANAEnterpriseID)
+		templateSet := createTemplateSet(templateID)
+		// Send template record
+		_, err = export.SendSet(templateSet)
 		if err != nil {
-			klog.Errorf("Did not find the element with name sourceIPv4Address")
+			klog.Fatalf("Got error when sending record: %v", err)
```
Comment (srikartati) [inline pkg/test/exporter_collector_test.go:291]: could we use t.Fatalf to fail the test?

File: pkg/test/exporter_collector_test.go [hunk 2]
```diff
@@ -260,172 +283,246 @@ func testExporterToCollector(address net.Addr, isMultipleRecord bool, isEncrypte
...
-		ie := entities.NewInfoElementWithValue(element, nil)
-		elements = append(elements, ie)
 
-		element, err = registry.GetInfoElement("destinationIPv4Address", registry.IANAEnterpriseID)
+		dataSet := createDataSet(templateID, isMultipleRecord)
+		// Send data set
+		_, err = export.SendSet(dataSet)
 		if err != nil {
-			klog.Errorf("Did not find the element with name destinationIPv4Address")
+			klog.Fatalf("Got error when sending record: %v", err)
```
Comment (srikartati) [inline pkg/test/exporter_collector_test.go:298]: same here.

File: pkg/exporter/process.go [hunk 3]
```diff
@@ -235,6 +235,7 @@ func (ep *ExportingProcess) createAndSendMsg(set entities.Set) (int, error) {
 	bytesSlice := append(msg.GetMsgBuffer().Bytes(), set.GetBuffer().Bytes()...)
 	// Send the message on the exporter connection.
 	bytesSent, err := ep.connToCollector.Write(bytesSlice)
+
```
Comment (srikartati) [inline pkg/exporter/process.go:238]: is this required? L237 and L238 belongs to same code block.

**Comment by srikartati** [comment]
Also, I presume this goes in to patch release v0.4.1. If so, it is better to create PR against release-0.4.0 branch and cherry pick the commit back to master. If there are any commits in the meanwhile on master that are not intended for patch release and have dependent code changes on the master, cherry-picking on to release branch that can be messy.

File: pkg/test/exporter_collector_test.go (hunk 1 above)
Comment (zyiou) [reply pkg/test/exporter_collector_test.go]: Sure. Updated.

File: pkg/exporter/process.go (hunk 3 above)
Comment (zyiou) [reply pkg/exporter/process.go]: it is not needed. removed.

**Comment by zyiou** [comment]
> Also, I presume this goes in to patch release v0.4.1. If so, it is better to create PR against release-0.4.0 branch and cherry pick the commit back to master. If there are any commits in the meanwhile on master that are not intended for patch release and have dependent code changes on the master, cherry-picking on to release branch that can be messy.
> 
> As there are no such commits in master, we are ok. Please consider this for future.

Thanks for reminding. will be more careful in the future.

**Comment by srikartati** [comment]
Keeping version file in a different commit is better. It will be easy to cherrypick this one on to master.

**Review by srikartati — APPROVED** [review]

**Pull request merged**